File: 225-用队列实现栈.cpp
class MyStack {
public:
    queue<int> myqueue;
    int length;

    /** Initialize your data structure here. */
    MyStack() {
        length = 0;
    }
    
    /** Push element x onto stack. */
    void push(int x) {
        myqueue.push(x);
        length += 1;
    }
    
    /** Removes the element on top of the stack and returns that element. */
    int pop() {
        for(int i = 0; i < length - 1; i++){
            myqueue.push(myqueue.front());
            myqueue.pop();
        }
        int result = myqueue.front();
        myqueue.pop();
        length -= 1;
        return result;
    }
    
    /** Get the top element. */
    int top() {
        for(int i = 0; i < length - 1; i++){
            myqueue.push(myqueue.front());
            myqueue.pop();
        }
        int result = myqueue.front();
        myqueue.push(result);
        myqueue.pop();
        return result;
    }
    
    /** Returns whether the stack is empty. */
    bool empty() {
        return length == 0;
    }
};

File: 225-用队列实现栈.py
from collections import deque

class MyStack:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.queue = deque()
        self.length = 0
        

    def push(self, x: int) -> None:
        """
        Push element x onto stack.
        """
        self.queue.append(x)
        self.length += 1

    def pop(self) -> int:
        """
        Removes the element on top of the stack and returns that element.
        """
        for i in range(self.length - 1):
            self.queue.append(self.queue.popleft())
        self.length -= 1
        return self.queue.popleft()
        

    def top(self) -> int:
        """
        Get the top element.
        """
        for i in range(self.length - 1):
            self.queue.append(self.queue.popleft())
        result = self.queue[0]
        self.queue.append(self.queue.popleft())
        return result
        

    def empty(self) -> bool:
        """
        Returns whether the stack is empty.
        """
        return self.length == 0
